### PR TITLE
feat: filters component

### PR DESCRIPTION
**Pull request by theogainey**
### Description
renamed OverviewFilters to Filters
add SelectControl to Filters component
add styles for Filters component

### Spec

Designs: [Designs](https://www.figma.com/file/kswpPSHH2h5njVyfGDbIHk/FSA-Capstone-Project-2022?node-id=0%3A1)

See Story: [ISSUE_NUMBER](https://sparkbox.atlassian.net/browse/FSA22V1-193)

### Validation
* [x] This PR has visual elements, so it was reviewed by a designer. 
* [x] This PR has code changes, and our linters still pass.
* [x] This PR affects production code, so it was browser tested (see below).
* [x] Changes are browser tested. This includes functionality, accessibility, responsiveness, and design.

#### To Validate

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, Snyk, etc).
2. Pull down all related branches.
3. Confirm all tests pass.
4. Browser test multiple viewport widths

### CODE REVIEW

File: components/SelectControl.jsx
Comment (mdt2) [inline components/SelectControl.jsx:13]: I've never used `htmlFor`, based on a quick search it looks like it's React's version of the `for` attribute, is that right?

File: styles/components/_filter-toggle.scss
Comment (mdt2) [inline styles/components/_filter-toggle.scss:33]: I know that this isn't new code, and this suggestion doesn't have to be done in this PR, but it can be nice to create a variable for transition times. So you could have `$transition-quick: 250ms` and that way you only have to change it in one place if the value ever needs adjusted.

File: pages/index.jsx
Comment (mdt2) [inline pages/index.jsx:19]: Can you help me understand the thought process that went into this change?

File: components/Layout.jsx
Comment (mdt2) [inline components/Layout.jsx:10]: Yeah, curious to know about this change. Given that the Filters aren't on every page, I probably wouldn't consider them part of the `<header>` since that piece is usually constant across the site, both for user experience and for accessibility.

**Comment by empee3** [comment]
This is looking great! I did run into a small issue on mobile, where it looks like the active state for the filters are sticking until another filter is selected. I've included a screen recording as a demo.

https://user-images.githubusercontent.com/2685062/172657816-d07f3a57-39b0-43f8-b133-17835cda0f27.mov

**Review by mdt2 — COMMENTED** [review]
This is looking good, Theo! I left a couple questions for you. The filters are definitely adjusting with the browser 🎉 Were you able to pair with a designer on what the filters should look like at smaller screen sizes? This is what I see currently for a phone size screen:

Screen Shot 2022-06-08 at 11 35 05 AM

File: components/SelectControl.jsx
Comment (theogainey) [reply components/SelectControl.jsx:13]: Yes it is! Just like `className` its one of those things you think React would be smart enough to be able to handle without the weird casing.

File: styles/components/_filter-toggle.scss
Comment (theogainey) [reply styles/components/_filter-toggle.scss:33]: Good idea! And even though it's not explicitly apart of this PR I think it makes sense for me to go ahead and include it here.

File: components/Layout.jsx
Comment (theogainey) [reply components/Layout.jsx:10]: The reasoning behind this change was so that the SkipToContent link will skip the filters and go straight to the content. Why I put it in the `<header>` because I was getting an axe dev tool warning about content being contained in landmarks. Now that you said something I'm thinking it would probably be more appropriate to include in a `<section>` and include some conditional logic so that the section only appears on the overview page

**Comment by theogainey** [comment]
> This is looking good, Theo! I left a couple questions for you. The filters are definitely adjusting with the browser 🎉 Were you able to pair with a designer on what the filters should look like at smaller screen sizes? This is what I see currently for a phone size screen:
> 
> Screen Shot 2022-06-08 at 11 35 05 AM

Yep I got it approved by Jelly. Originally I had a comment in the PR validation that I would check that off after I got approval. I probably should have left that in after I checked the box so people would know that I did do that.

File: components/Layout.jsx
Comment (mdt2) [reply components/Layout.jsx]: Oh yeah, good call skipping over the filters! I think that making this a prop might be an unnecessary layer of abstraction. Based on the designs it seems like the only component that will ever need to be skipped (other than the header) is the filters, is that right? So I think we can put the Filters component directly under the `<header>` wrapped in a `<section>` like you said so that the code is more readable for someone new to the code. Let me know what you think!

**Review by mdt2 — COMMENTED** [review]
One more thing to think about as you make the filter change on small screen. Getting really close 💯

File: components/Layout.jsx
Comment (mdt2) [reply components/Layout.jsx]: Going to resolve this! Per our zoom convo, we want to stay away from prop drilling and this solution doesn't require extra conditionals to turn filters on/off per page ✨

File: components/Layout.jsx
Comment (mdt2) [inline components/Layout.jsx:10]: Great rename!

**Review by mdt2 — APPROVED** [review]
Great work Theo!

File: util/getUniqueRoles.js
Comment (arnest00) [inline util/getUniqueRoles.js:10]: 🏆

**Review by arnest00 — APPROVED** [review]
Man, this looks so good

**Pull request merged**
